### PR TITLE
Add support for app-role authentication when using Vault as Secret Backend

### LINKED ISSUE #23: Add support for app-role authentication when using Vault as Secret Backend
Would be nice if it's possible to add approle authentication to Vault secret backend. 

Use something like the [hashi_vault](https://docs.ansible.com/ansible/latest/plugins/lookup/hashi_vault.html) lookup plugin of Ansible. This one will read out Env vars like:
- VAULT_ROLE_ID
- VAULT_SECRET_ID
- VAULT_AUTH_METHOD

I will try to create something, but not much experience with writing in Go

### PR DESCRIPTION
**Pull request by stokkie90**
Add support for approle authentication

Closes #23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,23 +163,37 @@ EOF
...
 
-- [Vault](#vault)
-- [AWS SSM Parameter Store](#aws-ssm-parameter-store)
-- [AWS Secrets Manager](#aws-secrets-manager)
-- [GCP Secrets Manager](#gcp-secrets-manager)
-- [SOPS](#sops) powered by [sops](https://github.com/mozilla/sops))
-- [Terraform (tfstate)](#terraform-tfstate) powered by [tfstate-lookup](https://github.com/fujiwara/tfstate-lookup)
-- [Echo](#echo)
-- [File](#file)
+- [vals](#vals)
```
Comment (mumoshu) [inline README.md]: I don't get the goal of this change. Perhaps you wanted to name this whole section like "ToC" and move it into the very beginning of this README? 😃

**Comment by mumoshu** [comment]
@stokkie90 Thanks for the PR! Please feel free to poke me once you think this is ready to be reviewed 😄

File: README.md (hunk 1 above)
Comment (stokkie90) [reply README.md]: My editor did that, reverted it.

**Comment by stokkie90** [comment]
@mumoshu It can be reviewed now

File: README.md (hunk 1 above)
Comment (mumoshu) [reply README.md]: @stokkie90 Gotcha! Btw, which editor was it? I'm interested because it seems useful if used correctly.

Comment (stokkie90) [reply README.md]: Used vscode. With a markdown extension.

Comment (mumoshu) [reply README.md]: Will try. Thanks for the info!

**Comment by stokkie90** [comment]
@mumoshu could you also create a new version maybe, Than i can update the version in Helmfile repo also to include this feature.

**Pull request merged**